packages/aws-cdk-lib/core/lib/private/unique-resource-name.ts

```diff
@@ -59,7 +67,7 @@ export function makeUniqueResourceName(components: string[], options: MakeUnique
   // in order to support transparent migration of cloudformation templates to the CDK without the
   // need to rename all resources.
   if (components.length === 1) {
-    const topLevelResource = removeNonAllowedSpecialCharacters(components[0], separator, options.allowedSpecialCharacters);
+    const topLevelResource = prefix + removeNonAllowedSpecialCharacters(components[0], separator, options.allowedSpecialCharacters);
```

Comment:
Shouldn't this prefix also be included in the `removeNonAllowedSpecialCharacters`?
```suggestion
    const topLevelResource = removeNonAllowedSpecialCharacters(prefix + components[0], separator, options.allowedSpecialCharacters);
```

Reply:
This is done to keep the new name as close as possible to the original behavior. When a shortened stack name had a prefix, the final result would be "Prefix-ShortenedStackName". If the prefix is included in the removal of special characters, the name generated would be "PrefixShortenedStackName".

Reply:
ah yeah you are right 👍